### ISSUE TITLE
Example with different approaches for error handling

### ISSUE THREAD

**Issue by marktani** [graphcool/prisma#1892]
There [has been some discussion](https://github.com/graphcool/prisma/issues/1738) about how to throw errors from resolvers properly.

An example that show cases different approaches that are considered best practice would be super helpful!

If anyone's interested in contributing such an example, feel free to ping me [in Slack](slack.graph.cool) (my handle there is @nilan), happy to collaborate on this one! 🙌

**Comment by akoenig** [comment]
I thought it might be a good idea to explain the `apollo-error` approach in more depth. That is why I compiled this blog post here: https://dev.to/andre/handling-errors-in-graphql--2ea3.

I'm happy to hear your thoughts :)

**Comment by morajabi** [comment]
@akoenig I do uncontrolled like that too. The beauty is, I can wire Sentry/Raven for capturing errors in the error handler or helmet as in your article.

**Comment by akoenig** [comment]
@morajabi Yeah, definitely. I do that too! Glad to see that this is moving to a common consensus 🙂💪

IMHO having a middleware layer is super powerful. You can compose small functions together and therefore keep your resolvers „clean“. 🙌🏽
